umbra-js/src/utils/utils.ts

```diff
@@ -283,9 +283,9 @@ function getResolutionInstance(provider: EthersProvider) {
   const networkName = provider.network.name === 'homestead' ? 'mainnet' : provider.network.name;
   return new Resolution({
     sourceConfig: {
-      cns: {
-        provider: Eip1993Factories.fromEthersProvider(provider),
-        network: networkName as 'mainnet' | 'rinkeby',
+      uns: {
+        api: true,
```

Comment:
Does this mean we rely on a UD-hosted API instead of reading directly from the chain? What are the other options/tradeoffs here?

Reply:
You don't have to rely on UD-hosted APIs, i modified the resolution to read from on-chain with Eip1993Factories provider with a new commit.